javaee-cafe/src/main/java/cafe/web/view/Cafe.java

```diff
@@ -31,86 +31,86 @@
-					+ FacesContext.getCurrentInstance().getExternalContext().getRequestServerPort()
-					+ "/javaee-cafe/rest/coffees";
-			this.client = ClientBuilder.newBuilder().hostnameVerifier(new HostnameVerifier() {
-				public boolean verify(String hostname, SSLSession session) {
-					return true;
-				}
-			}).build();
-			this.getAllCoffees();
-		} catch (IllegalArgumentException | NullPointerException | WebApplicationException | UnknownHostException ex) {
-			logger.severe("Processing of HTTP response failed.");
-			ex.printStackTrace();
-		}
-	}
-
-	private void getAllCoffees() {
-		this.coffeeList = this.client.target(this.baseUri).path("/").request(MediaType.APPLICATION_JSON)
-				.get(new GenericType<List<Coffee>>() {
-				});
-	}
-
-	public void addCoffee() {
-		Coffee coffee = new Coffee(this.name, this.price);
-		this.client.target(baseUri).request(MediaType.APPLICATION_JSON).post(Entity.json(coffee));
-		this.name = null;
-		this.price = null;
-		this.getAllCoffees();
-	}
-
-	public void removeCoffee(String coffeeId) {
-		this.client.target(baseUri).path(coffeeId).request().delete();
-		this.getAllCoffees();
-	}
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass().getName());
+
+    private String baseUri;
+    private transient Client client;
+
+    @Inject
+    private transient SecurityContext securityContext;
+
+    @NotNull
+    @NotEmpty
+    protected String name;
+    @NotNull
+    protected Double price;
+    protected List<Coffee> coffeeList;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Double getPrice() {
+        return price;
+    }
+
+    public void setPrice(Double price) {
+        this.price = price;
+    }
+
+    public List<Coffee> getCoffeeList() {
+        return coffeeList;
+    }
+
+    public String getLoggedOnUser() {
+        return securityContext.getCallerPrincipal().getName();
+    }
+
+    @PostConstruct
+    private void init() {
+        try {
+            HttpServletRequest request = (HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext()
+                    .getRequest();
+
+            InetAddress inetAddress = InetAddress.getByName(request.getServerName());
+
+            baseUri = FacesContext.getCurrentInstance().getExternalContext().getRequestScheme() + "://"
+                    + inetAddress.getHostName() + ":"
+                    + FacesContext.getCurrentInstance().getExternalContext().getRequestServerPort()
+                    + "/javaee-cafe/rest/coffees";
+            this.client = ClientBuilder.newBuilder().hostnameVerifier(new HostnameVerifier() {
+                public boolean verify(String hostname, SSLSession session) {
+                    return true;
+                }
+            }).build().register(new CafeRequestFilter());
```

Comment:
This is the only meaning change of this file, all other changes are formatting only.